akka-docs/src/main/paradox/scala/stream/stream-flows-and-basics.md

```diff
@@ -354,4 +354,45 @@ been signalled already – thus the ordering in the case of zipping is defined b
 
 If you find yourself in need of fine grained control over order of emitted elements in fan-in
 scenarios consider using `MergePreferred`, `MergePrioritized` or `GraphStage` – which gives you full control over how the
-merge is performed.
\ No newline at end of file
+merge is performed.
+
+# Actor Materializer Life-cycle
+
+An important aspect of working with streams and actors is understanding an `ActorMaterializer`'s life-cycle.
+The materializer is bound to the life-cycle of the `ActorRefFactory` it is created from, which in practice will
+be either an `ActorSystem` or `ActorContext` (when the materializer is created within an `Actor`).
+
+As a quick reminder, the usual way of creating an `ActorMaterializer` is to create it next to your `ActorSystem`,
+which likely is in a "main" class of your application:
+
+Scala
+:   @@snip [FlowDocSpec.scala]($code$/scala/docs/stream/FlowDocSpec.scala) { #materializer-from-system }
+
+Java
+:   @@snip [FlowDocTest.java]($code$/java/jdocs/stream/FlowDocTest.java) { #materializer-from-system }
+
+This case is simple and nothing suprising can happen here. We simply would run streams on this materializer,
+and if the system terminates, the materializer (and all of its streams) would be terminated as well. 
+
+If we look at the following example, where we create the `ActorMaterializer` within an `Actor`:
+
+Scala
+:   @@snip [FlowDocSpec.scala]($code$/scala/docs/stream/FlowDocSpec.scala) { #materializer-from-actor-context }
+
+Java
+:   @@snip [FlowDocTest.java]($code$/java/jdocs/stream/FlowDocTest.java) { #materializer-from-actor-context }
+
+In the above example we used the `ActorContext` to create the materializer. This binds its lifecycle to the surrounding `Actor`. In other words, while the stream we started there would under normal circumstances run forever, if we stop the Actor it would terminate the stream as well. We have *bound the streams' lifecycle to the surrounding actor's lifecycle*.
+This is a very useful technique if the stream is closely related to the actor, e.g. when the actor represents an user or other entity, that we continiously query using the created stream -- and it would not make sense to keep the stream alive when the actor has terminated already. The streams termination will be signalled by an "Abrupt termination exception" signaled by the stream.
+
+You may also cause an `ActorMaterializer` to shutdown by explicitly calling `shutdown()` on it, resulting in abruptly terminating all of the streams it has been running then. 
```

Comment:
perhaps also a word about that KillSwitch can be used to stop things, e.g. using a shared Materializer but stopping a stream bound to the actor via a KillSwitch from postStop.

Reply:
mentioned kill switch